src/Components/SavedRecipePage/SavedRecipesPage.tsx

```diff
@@ -1,49 +1,122 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { SavedRecipesPageProps } from '../../types';
-import SmallRecipeCard from '../SmallRecipeCard/SmallRecipeCard';
+import SavedRecipeCard from '../SavedRecipeCard/SavedRecipeCard';
 import Header from '../Header/Header';
-import ScrollToTopButton from '../ScrollToTopButton/ScrollToTopButton'
-import ReturnToSearchButton from '../ReturnToSearchButton/ReturnToSearchButton'
+import ScrollToTopButton from '../ScrollToTopButton/ScrollToTopButton';
+import ReturnToSearchButton from '../ReturnToSearchButton/ReturnToSearchButton';
 import { Recipe } from '../../types';
 import { getSavedRecipes } from '../apiCalls';
-import { useEffect, useState } from 'react';
 
-//why is it going to this pageeeE??? when we have a search input
-const SavedRecipesPage = () => {
-  // console.log('Saved Recipes:', savedRecipes)
-  const [ savedRecipes, setSavedRecipes ] = useState<Recipe[]>([])
+// ... (your imports)
+
+const SavedRecipesPage: React.FC<SavedRecipesPageProps> = ({ recipes, savedRecipes, updateSavedRecipes, updateSeletedFilteredRecipe }) => {
+  const id = 1;
 
   useEffect(() => {
-    getSavedRecipes() 
-    .then(savedRecipeData => {
-      setSavedRecipes(savedRecipeData.data.attributes.recipes)
-    }) 
-  })
-
-  const displaySavedRecipes = savedRecipes.map((savedRecipe) => (
-    <SmallRecipeCard
-      key={savedRecipe.name}
-      name={savedRecipe.name}
-      instructions={savedRecipe.instructions}
-      image_url={savedRecipe.image_url}
-      ingredients={savedRecipe.ingredients}
-      // isSaved={true}
-      // toggleSavedRecipes={() => toggleSavedRecipes(savedRecipe)}
-    />
-  ));
+    getSavedRecipes(id)
+      .then((savedRecipeData) => {
+        console.log('Received saved recipe data:', savedRecipeData);
+        const allSavedRecipes = savedRecipeData.data.attributes.recipes;
+        const userSpecificRecipes = allSavedRecipes.filter((recipe) => recipe.user_id === id);
+        console.log('User specific recipes:', userSpecificRecipes);
+        updateSavedRecipes(userSpecificRecipes);
+      });
+  }, [id]);
+
+  console.log('savedRecipes', savedRecipes);
+
+  // const uniqueRecipeFlat = savedRecipes.flat()
+  // console.log("uniqueRecipeFlat",uniqueRecipeFlat)
+
+  const uniqueRecipeNamesSet = new Set(savedRecipes.flat().map((recipe) => recipe.name));
```

Comment:
I'm going to have to have you break this down for me, I only understand it a little